code/game/machinery/computer/robot.dm

```diff
@@ -6,6 +6,9 @@
 	req_access = list(ACCESS_ROBOTICS)
 	circuit = /obj/item/circuitboard/computer/robotics
 	light_color = LIGHT_COLOR_PINK
+	var/mob/living/silicon/robot/locked_down_borg = null
```

Comment:
This will hard delete if the cyborg being locked down is deleted (gibbed / deconstructed), you will need reference handling (a signal to reset it on qdel, for example)